daft/delta_lake/delta_lake_scan.py

```diff
@@ -43,6 +44,24 @@ def __init__(
         deltalake_sdk_io_config = storage_config.config.io_config
         scheme = urlparse(table_uri).scheme
         if scheme == "s3" or scheme == "s3a":
+            # Try to get region from boto3
+            if deltalake_sdk_io_config.s3.region_name is None:
+                from botocore.exceptions import BotoCoreError
+
+                try:
+                    client = boto3_client_from_s3_config("s3", deltalake_sdk_io_config.s3)
+                    response = client.get_bucket_location(Bucket=urlparse(table_uri).netloc)
+                except BotoCoreError as e:
+                    logger.warning(
```

Comment:
Check out logger.exception for a more canonical way of doing this

Reply:
Oh shoot I totally read it as logger.error when you mentioned it before

Reply:
Hm actually it looks pretty similar. Is that something we'd want to do? I feel that I don't want it to be logged with error level.

Reply:
Exception is special, and gives a bunch of traceback which is useful.

> Benefits of using logger.exception():
> Automatic traceback logging: You don't need to manually format and log the traceback information.
> Consistent logging format: The logging format is standardized, making it easier to read and analyze logs.
> Error level: The message is logged at the ERROR level, which helps you identify and filter critical issues.